### ISSUE TITLE
Update for Deluge 2.0

### ISSUE THREAD

**Issue by keklol123** [cvarta/deluge-extractor#17]
Deluge 2.0.* is now the stable, any plan of updating this plugin to work with it?

Can't find a single working extractor, and as this one is the one ive used for years i would love for it to get an update.

**Comment by MoeJoe87** [comment]
Hi im also interested in an update for Deluge 2.0

**Comment by chrishuan9** [comment]
@keklol123 @MoeJoe87 , yes I'm working on fixing the issues and updating the plugin to make it work with deluge 2.0 - I am a bit strapped for time due to personal reasons but I will try my best.

**Comment by MoeJoe87** [comment]
Ok cool thank you. I tired the extractor from deluge itself...i thought aybe with an update its better....but its not really! 

The original extractor plugin reverts always my setting that it should create a torrent named subfolder! I dont understand why but after a while it is unthicked! Why is it so hard to leave the setting as the user want. ^^ I need to write the Deluge Devs. ^^

**Comment by chrishuan9** [comment]
@MoeJoe87 you can try to write to the Deluge Devs, as far as I know its a single guy doing all the development. The Development of version 2.0 was all his efforts

**Comment by Enf0** [comment]
Any progress on updating this for Deluge 2.0?

**Comment by spreadred** [comment]
https://deluge.readthedocs.io/en/latest/devguide/how-to/update-1.3-plugin.html - in case anyone is interested in the theoretical steps that might need to be taken.

Might be easier to just create this again from scratch?

**Comment by clutwo** [comment]
Any update on this?

**Comment by d8ahazard** [comment]
I've just submitted a pull request for V2 support! Done and done!

**Comment by danpowell88** [comment]
Are you able to provide the compiled plugin in the meantime before this gets merged?

**Comment by d8ahazard** [comment]
Yes, it's in the releases on my fork of the repository. 😁

**Comment by danpowell88** [comment]
@d8ahazard looks like its broken again, at least using the binhex docker container (https://hub.docker.com/r/binhex/arch-deluge)

Are you seeing the same issue, if I revert back to binhex/arch-deluge:2.0.3-1-02 everything seems to be fine again

**Comment by d8ahazard** [comment]
I saw that! Actually, I noticed it a few days ago, but I was zoned out on
some other project and didn't want to think about it then, so I just
reverted.

I took a minute today to examine the problem, and it seems like all I had
to do was rebuild the egg using Python 3.8. I'm still seeing some errors in
the log about logging, but the plugin shows up and seems to work.

There's a new release on my repo that's built with 3.8. ;)

Is @cvarta still alive? I'd love to see this make it to the regular
channels. :P
